src/gobject.cc

```diff
@@ -200,20 +197,30 @@ static void GObjectClassDestroyed(const Nan::WeakCallbackInfo<GType> &info) {
 
     DestroyVFuncs(gtype);
 
-    auto persistentTpl = (Persistent<FunctionTemplate> *)
+    auto persistentTpl = (Nan::Persistent<FunctionTemplate> *)
         g_type_get_qdata (gtype, GNodeJS::template_quark());
-    auto persistentFn  = (Persistent<Function> *)
+    auto persistentFn  = (Nan::Persistent<Function> *)
         g_type_get_qdata (gtype, GNodeJS::function_quark());
-    delete persistentTpl;
+    persistentTpl->Reset();
     delete persistentFn;
 
     g_type_set_qdata (gtype, GNodeJS::template_quark(), NULL);
     g_type_set_qdata (gtype, GNodeJS::function_quark(), NULL);
     g_free(gtypePtr);
 }
 
-static void GObjectFallbackPropertyGetter(Local<v8::Name> property,
-                                            const v8::PropertyCallbackInfo<Value>& info) {
+#if defined(V8_MAJOR_VERSION) && (V8_MAJOR_VERSION > 12 || \
+    (V8_MAJOR_VERSION == 12 && defined(V8_MINOR_VERSION) && V8_MINOR_VERSION > 4))
+#define PROPERTY_CALLBACK_RETURN_TYPE v8::Intercepted
+#define PROPERTY_CALLBACK_INFO_TYPE v8::PropertyCallbackInfo<void>
+#else
+#define PROPERTY_CALLBACK_RETURN_TYPE void
+#define PROPERTY_CALLBACK_INFO_TYPE v8::PropertyCallbackInfo<Value>
+#endif
```

Comment:
I've removed the return macros in favor of return Nan::Intercepted::(Yes|No), feels more readable.

The rest stays because [this commit message](https://github.com/romgrk/node-gtk/commit/6e4dfb74707773384e03c72fa74aa4fcb5708589) applies.